### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by katyo**
Solution for #73

Implemented tasks:
- [x] Check code formatting rules
- [x] Check code quality using `clippy`
- [x] Generate bindings for supported platforms (stable Rust)
- [x] Check building on all supported platforms (stable Rust)
- [x] Check building on different Rust channels (stable/beta/nightly, Linux x86_64)
- [x] Run tests on native Linux platforms (x86_64 and i686)
- [x] Create pull-requests for updating generated bindings
- [x] Publish crates when pushing version tag (It requires explicitly setting versions in manifests of crates)

All tasks runs on ubuntu 20.04.
System version of DRM headers is used, so bindings may lack some bleeding edge APIs.
To get publishing working you need set `CRATES_TOKEN` secret.

Currently supported platforms for `bindgen`:
- Linux x86
- Linux x86_64
- Linux arm
- Linux aarch64

Needed for [gbm.rs#4](/Smithay/gbm.rs/pull/4)

### CODE REVIEW

**Comment by katyo** [comment]
@Drakulix,
I would like to assign concrete version numbers to dependencies for some reasons.
I done it on another [CI branch](/katyo/drm-rs/compare/develop-ci...katyo:ci) to get working check with `-Z minimal-versions`.

**Comment by Drakulix** [comment]
> @Drakulix,
> I would like to assign concrete version numbers to dependencies for some reasons.
> I done it on another [CI branch](/katyo/drm-rs/compare/develop-ci...katyo:ci) to get working check with `-Z minimal-versions`.

I have filled #77

**Comment by katyo** [comment]
@Drakulix I have rebased this PR.
Is any more which should be done before merging?

**Comment by Drakulix** [comment]
> @Drakulix I have rebased this PR.
> Is any more which should be done before merging?

I do not think so.

@Slabity anything you would like to add?

**Comment by Drakulix** [comment]
@katyo Could you rebase one more time?

@Slabity I would merge this, setup the auto-publish feature for this repo and then release 0.4. Any objections?

**Comment by Slabity** [comment]
> @Slabity I would merge this, setup the auto-publish feature for this repo and then release 0.4. Any objections?

No objections at all. Having proper CI and auto-publishing is probably a good idea.

**Comment by katyo** [comment]
@Drakulix @Slabity Hope it's ready to merge.

Don't forget to add `CRATES_TOKEN` secret before publising on crates.io via tags.

File: .github/workflows/ci.yml
Comment (Drakulix) [inline .github/workflows/ci.yml]: Can you remove this, please?

File: .github/workflows/ci.yml
Comment (Drakulix) [inline .github/workflows/ci.yml:299]: Can this step deal with the three different packages of the repository?
Or do we need multiple steps here?

The order is important as well, since the crates depend on each other.
We need to push `drm-sys` first, then `drm-ffi` and at last `drm`.

**Review by Drakulix — APPROVED** [review]
Looks good!
Two last questions, through...

@Slabity can you setup the CRATES_TOKEN secret?
Alternatively I can do it, if you give me push access to `drm-sys` as well, otherwise the upload will fail.

File: .github/workflows/ci.yml
Comment (katyo) [reply .github/workflows/ci.yml:299]: This step uses [publish-crates](https://github.com/marketplace/actions/publish-crates) action which supports publishing workspace members with dependencies.
It works for my projects just a fine. I hope it will also works here.

File: .github/workflows/ci.yml
Comment (katyo) [reply .github/workflows/ci.yml]: Ok I remember, I remove it just before merge.

Comment (katyo) [reply .github/workflows/ci.yml]: Done

File: .github/workflows/ci.yml
Comment (MarijnS95) [inline .github/workflows/ci.yml:90]: @katyo You might want to expand this with `--all-targets` so that the examples are linted, too.

In other news something odd is going on with the `clippy` on `nightly`. The current 1.51 stable gives off extra `upper_case_acronyms` warnings in `src/lib.rs`, and tons of `redundant_field_names` warnings, but nightly doesn't. Is this a temporary fluke or do we need extra configuration? Have you seen this on other projects?

Comment (MarijnS95) [reply .github/workflows/ci.yml:90]: Ah, these crates are not in a workspace :)

**Pull request merged**
